Batons and Prods now require powercells, Fixes #5720

AtnerNT opened the pull request:
### Purpose
Like the title says, stun batons, stun prods and teleprods now will require power cells to power them, giving them a limited amount of charges before they must be replenished at a charging station.
Stun prods and teleprods also no longer require a powercell to craft, but one must be installed after the fact.
Power cells can be taken out and exchanged using a screw driver.
Power cells whose capacity is too small will not be able to be installed into the device.
At base level without modifying the power cell, stun batons can go 10 uses before running out, and the stun prod and teleprod can go 5 (Same capacity but twice the usage)

Fixes #5720 

CL: [Balance] stun batons and prods now need power cells with charge to function.

AtnerNT committed:
Powercells for batons

AtnerNT committed:
Hopefully that solves that

AtnerNT committed:
Update packages-lock.json

corp-0 commented on UnityProject/Assets/Scripts/Core/Editor/Items/MeleeEffectEditor.cs:
```diff
@@ -1,5 +1,6 @@
 using UnityEditor;
 using Weapons;
+using UnityEngine;
```
Uhm I wonder why did we do a custom inspector for this

AtnerNT replied:
MeleeEffect has quite a few fields that clog up the inspector, and not all of them are used all the time. So I made a custom inspector to just hide fields when they aren't needed so the inspector isn't as crowded with values.

AtnerNT replied:
(i.e teleport config options are hidden if the weapon cannot teleport)

NoooneyDude replied:
Might be before `NaughtyAttributes` was added.

AtnerNT replied:
It was not- however I only learnt that we had naughty attributes on this project then- Ill make the change now

AtnerNT committed:
Some small changes

Stun prods and Teleprods no longer require cells to craft and will not start with them. Cells with a smaller capacity then the min needed to use can no longer be installed.

AtnerNT committed:
Merge remote-tracking branch 'upstream/develop' into ProdandBatonCells

AtnerNT committed:
Changes from coustom inspector to NaughtyAttributes

corp-0 commented on UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:
```diff
@@ -9,30 +10,40 @@ namespace Weapons
 		public bool avoidImpassable = true;
 
-		[HideInInspector]
+		[ShowIf("canTeleport")]
```
change all the ``ShowIf`` to use ``nameof()`` instead of a string so if the field name changes in the future, the code doesn't silently stop working

corp-0 replied:
in this case it would be ``ShowIf(nameof(canTeleport))``

AtnerNT committed:
uses nameof() instead of string
